nussl/separation/nmf_mfcc.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import sklearn.cluster
+import librosa
+import nussl.audio_signal
+import nussl.constants
+import nussl.spectral_utils
+import nussl.utils
+from nussl.transformers import transformer_nmf
+from nussl.separation import separation_base
+import mask_separation_base
+import masks
+
+
+class NMF_MFCC(mask_separation_base.MaskSeparationBase):
+    """
+        Non Negative Matrix Factorization using K-Means Clustering on MFCC (NMF MFCC) is a source separation
+        algorithm that runs Transformer NMF on the magnitude spectrogram of an input audio signal.
+        It uses K means clustering to cluster the templates and activations returned by the NMF. The dot product of the
+        clustered templates and activations results in a magnitude spectrogram only containing a separated source.
+        This is used to create a Binary Mask object, and the whole process can be applied for each cluster to
+        return a list of Audio Signal objects corresponding to each separated source.
+
```

Comment:
There should also be a discussion about the relationship between this `TransformerNMF`. What is the distinction? How is it used?